### DISCUSSION TITLE
AudioPlayer: Switch from MP3 to AAC decoder

### DISCUSSION THREAD

**Discussion by Efraysse** [pschatzmann/arduino-audio-tools#1615] [Q&A] [answered]
Hello,

I try to use the audioPlayer class to read an AAC stream and next a MP3 stream : 

_const char *liste[] = { "http://icecast.radiofrance.fr/fip-hifi.aac","http://radiobonheur.ice.infomaniak.ch/radiobonheur-128-1.mp3" };_

I first declare the two decoders and start initialyse the player with th AAC decoder :

_MP3DecoderHelix decomp3;
AACDecoderHelix decoaac;
URLStream url("Livebox-18FA","XXXXXXXXXX");  
AudioSourceURL source(url, liste, "audio/aac");
SPDIFOutput out; 
AudioPlayer player(source,out,decoaac);_

In the loop section I catch an interruption to switch to the next stream (mp3) and so set the decoder to MP3: 

_player.setDecoder(decomp3);
source.nextStream(1);_

I think I am missing something because the first AAC stream play well but when switching to the other I get :

l_ibhelix - CommonHelix not active
libhelix - CommonHelix not active
[I] StreamCopy.h : 353 - try write  - 2 (open 1024 bytes) 
libhelix - CommonHelix not active
[I] StreamCopy.h : 353 - try write  - 3 (open 1024 bytes) 
libhelix - CommonHelix not active
[I] StreamCopy.h : 353 - try write  - 4 (open 1024 bytes) 
libhelix - CommonHelix not active
[I] StreamCopy.h : 353 - try write  - 5 (open 1024 bytes)_ 
...

**Comment by pschatzmann** [comment, marked as answer]
'CommonHelix not active' indicates that it has not been opened yet: I would expect that you would need to call begin()
